pkg/runs/runsDownload_test.go

```diff
@@ -167,41 +260,61 @@ func WriteMockRasRunsFilesResponse(
 	writer http.ResponseWriter,
 	req *http.Request,
 	desiredContents string) {
-	
-	writer.Header().Set("Content-Disposition", "attachment")	
+
+	writer.Header().Set("Content-Disposition", "attachment")
 	writer.Write([]byte(desiredContents))
 }
 
 // Creates a new mock server to handle requests from test methods
 func NewRunsDownloadServletMock(
 	t *testing.T,
 	status int,
-	runId string,
 	runName string,
-	artifactList []MockArtifact,
-	runResultStrings []string) *httptest.Server {
+	runResultStrings []string,
+	runs map[string][]MockArtifact) *httptest.Server {
 
 	server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
 
 		acceptHeader := req.Header.Get("Accept")
-		switch req.URL.Path {
-			case "/ras/runs":
-				assert.Equal(t, "application/json", acceptHeader, "Expected Accept: application/json header, got: %s", acceptHeader)
-				WriteMockRasRunsResponse(t, writer, req, runName, runResultStrings)
-
-			case fmt.Sprintf(`/ras/runs/%s/artifacts`, runId):
-				assert.Equal(t, "application/json", acceptHeader, "Expected Accept: application/json header, got: %s", acceptHeader)
-				WriteMockRasRunsArtifactsResponse(t, writer, req, artifactList)
-		}
-
-		runsFilesEndpoint := fmt.Sprintf(`/ras/runs/%s/files`, runId)
-		if strings.HasPrefix(req.URL.Path, runsFilesEndpoint) {
-			for _, artifact := range artifactList {
-				if req.URL.Path == (runsFilesEndpoint + artifact.path) {
-					WriteMockRasRunsFilesResponse(t, writer, req, artifact.path)
+		if req.URL.Path == "/ras/runs" {
+			assert.Equal(t, "application/json", acceptHeader, "Expected Accept: application/json header, got: %s", acceptHeader)
+			WriteMockRasRunsResponse(t, writer, req, runName, runResultStrings)
+
+		} else {
+			for runId, artifacts := range runs {
+				if req.URL.Path == "/ras/runs/"+runId+"/artifacts" {
+					assert.Equal(t, "application/json", acceptHeader, "Expected Accept: application/json header, got: %s", acceptHeader)
+					WriteMockRasRunsArtifactsResponse(t, writer, req, artifacts)
+				} else {
+					runsFilesEndpoint := fmt.Sprintf(`/ras/runs/%s/files`, runId)
+					if strings.HasPrefix(req.URL.Path, runsFilesEndpoint) {
+						for _, artifact := range artifacts {
+							if req.URL.Path == (runsFilesEndpoint + artifact.path) {
+								WriteMockRasRunsFilesResponse(t, writer, req, artifact.path)
+							}
+						}
+					}
 				}
 			}
 		}
+		// switch req.URL.Path {
```

Comment:
Remove all this commented-out code pls. It looks messy. If we don't need to run it, we don't need it.